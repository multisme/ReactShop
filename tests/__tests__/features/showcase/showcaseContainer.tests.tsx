import configureStore from 'redux-mock-store';
import * as redux from 'react-redux';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router} from 'react-router-dom';

import ShowcaseContainer from "features/showcase/showcaseContainer"

const mockStore = configureStore([]);

describe("showcaseContainer", () => {
        let store: any;
                  beforeEach(() => {
                          store = mockStore({
            })
        })

        it("render title on if selected is null", () => {
                        const mock = jest.spyOn(redux, 'useSelector');
                        mock.mockImplementation(() => ({ 
                                selected: null
                        }));
                        const component = mount(
                                <Router>
                                <Provider store={store}>
                                        <ShowcaseContainer/>
                                </Provider>
                                </Router>
                        );
                        const title = component.find('.header');
                        expect(title).toHaveLength(1);
                        expect(component).toMatchSnapshot();
        })
        it(" does render an item if selected is not null", () => {
                        const mock = jest.spyOn(redux, 'useSelector');
                        mock.mockImplementation(() => ({ 
                               selected : {id: 2, name: 3, quantity: 6, price: 8}
                        }));
                        const component = mount(
                                <Router>
                                <Provider store={store}>
                                        <ShowcaseContainer/>
                                </Provider>
                                </Router>
                        );
                        expect(component.find(".showcaseItem")).toHaveLength(1);
                        expect(component.find(".price")).toHaveLength(1);
                        expect(component.find(".name")).toHaveLength(1);
                        expect(component.find(".showcasePicture")).toHaveLength(1);
        })
})
