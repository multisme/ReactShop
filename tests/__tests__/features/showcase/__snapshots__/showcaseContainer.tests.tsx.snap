// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showcaseContainer render title on if selected is null 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <ShowcaseContainer>
        <div
          className="showcaseContainer green-border flex-centered"
        >
          <ShowcaseTitle>
            <div
              className="header"
            >
              <section>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
                <p>
                  THIS IS A SHOPPING WEBSITE
                </p>
              </section>
            </div>
          </ShowcaseTitle>
        </div>
      </ShowcaseContainer>
    </Provider>
  </Router>
</BrowserRouter>
`;
